src/runtime/arc.ts

```diff
@@ -175,10 +175,6 @@ constructor({id, context, pecFactories, slotComposer, loader, storageKey, storag
     }
 
     DriverFactory.unregister(this.volatileStorageDriverProvider);
-
-    for (const store of this._stores) {
-      Runtime.getRuntime().unregisterStore(store.id, [...this.findStoreTags(store)]);
```

Comment:
Is removing this ok? I worry that never unregistering stores will lead to data leaks (from a memory utilization perspective). I don't have much of the history around this code, perhaps @galganif can comment when he is in.

Reply:
we don't currently distinguish between arc that is being torn down (and can be spun up again) and deleted (no longer exists).
in the former case, we still want its stores to be available to other arcs (for example, if there is an ingestion arc that is only actively running when a new data comes in and then goes down again, we want other arcs to have access to the ingested data unrelated).
the latter case isn't supported at the moment (and needs to be as part of the upcoming better defined arc's lifecycle).